src/IECore/StreamIndexedIO.cpp

```diff
@@ -1941,6 +2227,57 @@ Imf::Int64 StreamIndexedIO::Index::writeUniqueData( const char *data, size_t siz
 	return loc;
 }
 
+StreamIndexedIO::Index::WriteInfo StreamIndexedIO::Index::writeUniqueDataCompressed( const char *data, size_t size, bool prefixSize )
+{
+	WriteInfo writeInfo;
+
+	int compressedSize = 0;
+
+	// TODO compress into multiple blocks which blosc can compress
```

Comment:
I think there's a good case for doing this now rather than later. If we're relying on compression to keep file sizes down, it's going to be very inconvenient to suddenly go _up_ a cliff when we go over this threshold. Would adding this in later also mean needing a second change of file format version?